web/files/files_test.go

```diff
@@ -0,0 +1,125 @@
+package files
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+var ts *httptest.Server
+var instance *middlewares.Instance
+
+func injectInstance(instance *middlewares.Instance) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("instance", instance)
+	}
+}
+
+func createDir(t *testing.T, path string) *http.Response {
+	res, err := http.Post(ts.URL+path, "text/plain", strings.NewReader(""))
+	assert.NoError(t, err)
+	return res
+}
+
+func upload(t *testing.T, path, contentType, body string) *http.Response {
+	buf := strings.NewReader(body)
+	res, err := http.Post(ts.URL+path, contentType, buf)
+	assert.NoError(t, err)
+	return res
+}
+
+func TestCreateDirWithNoType(t *testing.T) {
+	res := createDir(t, "/files/123")
+	assert.Equal(t, 422, res.StatusCode)
+	res.Body.Close()
+}
+
+func TestCreateDirWithNoName(t *testing.T) {
+	res := createDir(t, "/files/123?Type=io.cozy.folders")
+	assert.Equal(t, 422, res.StatusCode)
+	res.Body.Close()
+}
+
+func TestCreateDirOnNonExistingParent(t *testing.T) {
+	res := createDir(t, "/files/noooop?Name=foo&Type=io.cozy.folders")
+	assert.Equal(t, 404, res.StatusCode)
+	res.Body.Close()
+}
+
+func TestCreateDirAlreadyExists(t *testing.T) {
+	res := createDir(t, "/files/123?Name=456&Type=io.cozy.folders")
+	assert.Equal(t, 409, res.StatusCode)
+	res.Body.Close()
+}
+
+func TestCreateDirSuccess(t *testing.T) {
+	res := createDir(t, "/files/123?Name=coucou&Type=io.cozy.folders")
+	assert.Equal(t, 201, res.StatusCode)
+	res.Body.Close()
+
+	storage, _ := instance.GetStorageProvider()
+	exists, err := afero.DirExists(storage, "123/coucou")
+	assert.NoError(t, err)
+	assert.True(t, exists)
+}
+
+func TestCreateDirWithSlashInName(t *testing.T) {
+	res := createDir(t, "/files/123?Name=coucou/les/copains!&Type=io.cozy.folders")
```

Comment:
I am not sure we want to allow creation of a folder with "/" in names as it will not work when mapping it to an actual FS (like cozy-desktop does). We should probalby just throw a 400 "no slash in name".